libbeat/autodiscover/autodiscover_test.go

```diff
@@ -239,6 +238,46 @@ func TestAutodiscover(t *testing.T) {
 	assert.True(t, runners[1].started)
 	assert.False(t, runners[1].stopped)
 
+	//Test a start event that has no changes
+	eventBus.Publish(bus.Event{
+		"id":       "foo",
+		"provider": "mock",
+		"start":    true,
+		"meta": common.MapStr{
+			"foo": "baz",
+		},
+	})
+	wait(t, func() bool { return len(adapter.Runners()) == 2 })
+	runners = adapter.Runners()
+	assert.Equal(t, len(runners), 2)
+	assert.Equal(t, len(autodiscover.configs["mock:foo"]), 1)
+	assert.True(t, runners[0].stopped)
+	assert.Equal(t, runners[1].meta.Get()["foo"], "baz")
+	assert.True(t, runners[1].started)
+	assert.False(t, runners[1].stopped)
+
+	// Test a start event again that has meta changes
+	eventBus.Publish(bus.Event{
+		"id":       "foo",
+		"provider": "mock",
+		"start":    true,
+		"meta": common.MapStr{
+			"foo": "baz",
+			"abc": "def",
+		},
+	})
+	// wait() wouldn't work here as the runners are already 2 and a relist type event
+	// would just try to stop start the runner. Hence we sleep here.
+	time.Sleep(time.Second)
```

Comment:
Perhaps you could wait to metadata being available?

```suggestion
	wait(t, func() bool { runners := adapter.Runners(); return runners[1].meta.Get()["abc"] == "def" })
```

Reply:
done. thanks